zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceTest.java

```diff
@@ -1091,6 +1094,164 @@ public void shouldUnsubscribeUnmappedMessageBoundaryEvent() {
         .isTrue();
   }
 
+  @Test
+  public void shouldWriteMigratedEventWhenActiveElementHasAJobIncident() {
+    // given
+    final String sourceProcessId = "process";
+    final String targetProcessId = "process2";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .serviceTask("A", t -> t.zeebeJobType("jobTypeA"))
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .serviceTask("B", t -> t.zeebeJobType("jobTypeB"))
+                    .endEvent("target_process_end")
+                    .done())
+            .deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(sourceProcessId).create();
+
+    ENGINE.jobs().withType("jobTypeA").withMaxJobsToActivate(1).activate();
+    RecordingExporter.jobRecords(JobIntent.CREATED).withType("jobTypeA").await();
```

Comment:
🔧 No need to await for the job `CREATED` event, because the test job client takes care of this if you don't provide an explicit job key

https://github.com/camunda/zeebe/blob/99a058d7fd242199f94c715ef27038d55cf5c03c/zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/client/JobClient.java#L146-L147

https://github.com/camunda/zeebe/blob/99a058d7fd242199f94c715ef27038d55cf5c03c/zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/client/JobClient.java#L123-L136